### PR TITLE
Add update schematics for constructor changes

### LINKED ISSUE #12198: Provide upgrade schematics (product-configurator) for constructor deprecations
Cover changes to group menu component (if we implement that for 4.0) as well

### PR DESCRIPTION
**Pull request by ChristophHi**
Adds update schematics for configurator related constructor changes, for the update to 4.0.

Closes #12198

### CODE REVIEW

File: docs/migration/4_0.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
 # Technical Changes in Spartacus 4.0
 
 ## Breaking Changes Introduced in 4.0
+
+`ConfiguratorAttributeCheckboxListComponent` now also requires `ConfiguratorAttributeQuantityService`.
+`ConfiguratorAttributeDropDownComponent` now also requires `ConfiguratorAttributeQuantityService`.
+`ConfiguratorAttributeRadiButtonComponent` now also requires `ConfiguratorAttributeQuantityService`.
+`ConfiguratorCartEntryInfoComponent` now also requires `CommonConfiguratorUtilsService`.
```
Comment (Platonn) [inline docs/migration/4_0.md]: For every class TS source code, can we please:
- make the depdendency required
- drop deprecated constructor declarations?

FYI Sorry I didn't notice some of work might have been already done in https://github.com/SAP/spartacus/issues/10946 (and PR https://github.com/SAP/spartacus/pull/12355). Please review it not to duplicate the work.

Comment (ChristophHi) [reply docs/migration/4_0.md]: But I recall a 3.4 release was still planned? Then we cannot do those changes now, right?

File: projects/schematics/src/migrations/4_0/constructor-deprecations/data/configurator-storefront-utils.service.migration.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  importPath: SPARTACUS_PRODUCT_CONFIGURATOR_RULEBASED,
+  deprecatedParams: [
+    {
+      className: CONFIGURATOR_GROUPS_SERVICE,
+      importPath: SPARTACUS_PRODUCT_CONFIGURATOR_RULEBASED,
+    },
+    {
+      className: PLATFORM_ID_STRING,
+      importPath: ANGULAR_CORE,
+    },
```
Comment (Platonn) [inline projects/schematics/src/migrations/4_0/constructor-deprecations/data/configurator-storefront-utils.service.migration.ts:24]: `@inject` decorator is handled special way
```suggestion
    {
      className: PLATFORM,
      literalInference: ANY_TYPE,
      injectionToken: {
        token: PLATFORM_ID_STRING,
        importPath: ANGULAR_CORE,
      },
    },
```
(might need to run prettier)

Comment (ChristophHi) [reply projects/schematics/src/migrations/4_0/constructor-deprecations/data/configurator-storefront-utils.service.migration.ts:24]: No, this does not work. I also found this technique but saw that the constructor-deprecations mechanism in this case does not find the constructors it needs to update. Please see branch https://github.com/SAP/spartacus/tree/feature/GH-12198-testsForConstructorDeprecations, and run `yarn test `in the schematics folder

**Review by Platonn — DISMISSED** [review]

**Comment by ChristophHi** [comment]
--

**Review by marlass — DISMISSED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**